src/flb_upstream.c

```diff
@@ -84,13 +92,42 @@ struct flb_upstream *flb_upstream_create(struct flb_config *config,
     /* Set default networking setup values */
     flb_net_setup_init(&u->net);
 
-    u->tcp_host      = flb_strdup(host);
+    /* Set upstream to the http_proxy if it is specified. */
+    if (config->http_proxy) {
+        flb_debug("[upstream] config->http_proxy: %s", config->http_proxy);
+        ret = flb_utils_proxy_url_split(config->http_proxy, &proxy_protocol, &proxy_username, &proxy_password, &proxy_host, &proxy_port);
```

Comment:
this line exceeds 90 chars: https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#line-length

Reply:
Fixed.